### ISSUE TITLE
[reporter] Stuck after reporter Run is canceled by context

### ISSUE THREAD

**Issue by at15** [xephonhq/xephon-b#43]
got stuck before finalize reporter, for counter reporter, there is nothing to finalize, should be some go routine problem on my end ...

**Comment by at15** [comment]
met it again when using akumuli ...

````
INFO 0000 target database is akumuli_0 type akumuli
INFO 0000 workload is workload_0 series 1 value generator is constant
INFO 0000 reporter counter_0 is type counter
INFO 0000 counter reporter start 2018-03-18 18:50:22.688795797 -0700 PDT m=+0.022608676
INFO 0000 worker 3 started
INFO 0000 worker 7 started
INFO 0000 worker 6 started
INFO 0000 worker 4 started
INFO 0000 worker 8 started
INFO 0000 worker 0 started
INFO 0000 worker 9 started
INFO 0000 worker 5 started
INFO 0000 worker 1 started
INFO 0000 worker 2 started
INFO 0010 worker 8 exit due to context
INFO 0010 worker 0 exit due to context
INFO 0010 worker 2 exit due to context
INFO 0010 worker 3 exit due to context
INFO 0010 worker 9 exit due to context
INFO 0010 counter reporter stopped by context
INFO 0010 counter reporter stop 2018-03-18 18:50:32.689397978 -0700 PDT m=+10.023210745 duration 10.000602069s
````

**Comment by at15** [comment]
solved in #45 , need to keep in mind when using both context can channel, the reader should not use same context as writer, otherwise, the writer are pretty likely to be blocked ....
